### PR TITLE
Adding short tutorials showing how to use the new h2o.sklearn module

### PR DESCRIPTION
**Pull request by sebhrusen**
the 2 Jupyter notebooks can be viewed under https://github.com/h2oai/h2o-tutorials/tree/sklearn-support/tutorials/sklearn-integration

There is a generic one explaining in details how the new `h2o.sklearn` wrappers can be used in combination with Scikit-learn components, and summing-up how this works.

the second one is more specifically dedicated to `AutoML`, adding some examples.

### CODE REVIEW

**Comment by hannah-tillman** [comment]
@sebhrusen 
Just an FYI: when I'm running the first example ("mixing sklearn with h2o.sklearn components") for the "H2O-3 integration with Scikit-Learn" file, I get this error when running line "In [2]" when I am using Python 3. This error does not occur when running the code with Python 2.
`...
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/urllib/request.py in do_open(self, http_class, req, **http_conn_args)
   1317                           encode_chunked=req.has_header('Transfer-encoding'))
   1318             except OSError as err: # timeout error
-> 1319                 raise URLError(err)
   1320             r = h.getresponse()
   1321         except:

URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1056)>`

File: tutorials/sklearn-integration/H2OAutoML_as_sklearn_estimator.ipynb [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+    "\n",
+    "Like for other `H2O-3` estimators (see `H2O_estimators_as_sklears_estimators` notebook in the current folder for more details), the new `h2o.sklearn` module exposes 2 wrappers for `H2OAutoML`:\n",
+    "- `H2OAutoMLClassifier`\n",
+    "- `H2OAutoMLRegressor`\n",
+    "\n",
+    "Those wrappers expose the standard API familiar to `sklearn` users (`fit`, `predict`, `fit_predict`, `score`, `get_params`, `set_params`) and accept various format as input data (`H2OFrame`, `numpy` array, `pandas` Dataframe) which allow them to be combined with pure `sklearn` components in pipelines.\n",
+    "\n",
+    "Finally, user can still access the original `H2OAutoML` estimator from the wrapper thanks to its `estimator` property: this gives access to all the methods and properties declared in `H2OAutoML`, like `leaderboard`, `leader`, `event_log`, `training_info`, ...\n",
+    "\n",
+    "Again, for more details, it is recommended to have a look at the `H2O_estimators_as_sklears_estimators` notebook in this folder."
```
Comment (hannah-tillman) [inline tutorials/sklearn-integration/H2OAutoML_as_sklearn_estimator.ipynb]: 'H2O_estimators_as_**sklears**_estimators' <- spelling mistake

File: tutorials/sklearn-integration/h2o_estimators_as_sklearn_estimators.ipynb [hunk 2]
```diff
@@ -0,0 +1,1465 @@
...
+    "  - `transform(X)`, `fit_transform(X, y)`, `inverse_transform(X)` for all transformers, and also estimators supporting transformations (e.g. H2OPrincipalComponentAnalysisEstimator).\n",
+    "  - `score(X, y)` for estimators, using, like in `sklearn`, `sklearn.metrics.accuracy_score` for classifiers, and `sklearn.metrics.r2_score` for regressors.\n",
+    "- also expose an `estimator` property (since `3.28.0.2`; in `3.28.0.1` it is only available as the `_estimator` \"private\" property) which is a reference to the original `H2OEstimator` (or `H2OAutoML`) instance: this way, the user can still access properties and methods from the estimator that are not exposed directly by the wrapper.\n",
+    "- data parameters `X` and `y` accept various kind of data: `H2OFrame`, `numpy` arrays, `pandas.DataFrame`. The wrapper will try to return predictions or transformation in the same format as the input. Basic rules will be explained below, but first, this means that those wrappers can be chained with any kind of `sklearn` component in a `sklearn.pipeline.Pipeline`.\n",
```
Comment (hannah-tillman) [inline tutorials/sklearn-integration/h2o_estimators_as_sklearn_estimators.ipynb]: 'various **kind** of data:' <- "kinds" (small grammatical change)

File: tutorials/sklearn-integration/h2o_estimators_as_sklearn_estimators.ipynb [hunk 3]
```diff
@@ -0,0 +1,1465 @@
...
+   "metadata": {},
+   "source": [
+    "### Demo"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that you understand how those `H2O` wrappers work and how they combine with `sklearn` components, we can use enhance `H2O` with some unique `skklearn` features."
```
Comment (hannah-tillman) [inline tutorials/sklearn-integration/h2o_estimators_as_sklearn_estimators.ipynb]: 'skklearn' <- spelling mistake

**Comment by sebhrusen** [comment]
@hannah-tillman thanks for the review, I will fix the mistakes.
regarding the `URLError`, I was using Python 3 and didn't get the error. As it's using `iris` though, I'll remove the dependency to our dataset stored on s3 and use the `iris` dataset provided by `sklearn` instead: it's just too bad that the latter is already encoded, I also wanted to show that contrary to `sklearn`, with H2O we can still use non-encoded datasets...

File: tutorials/sklearn-integration/H2OAutoML_as_sklearn_estimator.ipynb [hunk 4]
```diff
@@ -0,0 +1,529 @@
...
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Parse progress: |█████████████████████████████████████████████████████████| 100%\n",
+      "Parse progress: |█████████████████████████████████████████████████████████| 100%\n",
+      "AutoML progress: |██████████\n",
+      "17:56:49.537: Skipping training of model GBM_5_AutoML_20200121_175640 due to exception: water.exceptions.H2OModelBuilderIllegalArgumentException: Illegal argument(s) for GBM model: GBM_5_AutoML_20200121_175640.  Details: ERRR on field: _min_rows: The dataset size is too small to split for min_rows=100.0: must have at least 200.0 (weighted) rows, but have only 112.0.\n",
```
Comment (ledell) [inline tutorials/sklearn-integration/H2OAutoML_as_sklearn_estimator.ipynb]: @sebhrusen Can we use a bigger dataset that does not produce errors?

Comment (sebhrusen) [reply tutorials/sklearn-integration/H2OAutoML_as_sklearn_estimator.ipynb]: just replaced `iris` with `breast-cancer` dataset example from `sklearn`: no error and better perf of SEs

**Comment by sebhrusen** [comment]
@hannah-tillman , @ledell are we ok with merging this?

**Pull request merged**